src/fs_io.cc

```diff
@@ -133,35 +152,18 @@ diskCombineWrites(_fde_disk *fdd)
      */
 
     if (fdd->write_q != nullptr && fdd->write_q->next != nullptr) {
-        int len = 0;
+        size_t wantCapacity = 0;
 
         for (dwrite_q *q = fdd->write_q; q != nullptr; q = q->next)
-            len += q->len - q->buf_offset;
-
-        dwrite_q *wq = (dwrite_q *)memAllocate(MEM_DWRITE_Q);
-
-        wq->buf = (char *)xmalloc(len);
-
-        wq->len = 0;
-
-        wq->buf_offset = 0;
+            wantCapacity += q->len - q->buf_offset;
```

Comment:
```suggestion
            wantCapacity += q->len - q->buf_offset; // XXX: might overflow
```

I do not insist on this PR fixing that old problem.